templates/commands/goldentest/new_test_cli.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package goldentest implements golden test related subcommands.
+package goldentest
+
+// This file implements the "templates golden-test new-test" subcommand.
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/abc/templates/common/input"
+	"github.com/abcxyz/abc/templates/common/specutil"
+	"github.com/abcxyz/abc/templates/common/templatesource"
+	"github.com/abcxyz/abc/templates/model/decode"
+	goldentest "github.com/abcxyz/abc/templates/model/goldentest/v1beta3"
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+)
+
+type NewTestCommand struct {
+	cli.BaseCommand
+
+	flags NewTestFlags
+}
+
+func (c *NewTestCommand) Desc() string {
+	return "create a new golden test"
+}
+
+func (c *NewTestCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <test_name> <location>
```

Comment:
Idea: maybe location should default to the current directory if not specified?

Reply:
done. I've changed the location from positional argument to optional argument. Let me know if it is what you were thinking?